packages/block-library/src/navigation/index.php

```diff
@@ -85,6 +85,41 @@ function block_core_navigation_build_css_font_sizes( $attributes ) {
 	return $font_sizes;
 }
 
+/**
+ * Renders a Navigation Block derived from data from the theme_location assigned
+ * via the block attribute 'location'.
+ *
+ * If the theme doesn't explicity support 'block-nav-menus' or no location was provided
+ * as a block attribute then an empty string is returned.
+ *
+ * @param  array $attributes Navigation block attributes.
+ * @return string HTML markup of a generated Navigation Block.
+ */
+function get_classic_navigation_elements( $attributes ) {
+
+	if ( ! current_theme_supports( 'block-nav-menus' ) ) {
+		return '';
+	}
+
+	if ( ! array_key_exists( 'location', $attributes ) ) {
+		return '';
+	}
+
+	$block_attributes = $attributes;
+	unset( $block_attributes['location'] );
```

Comment:
I don't follow the swapping values and the unset. It's probably related to the updates in `wp_nav_menu` but at least it needs a comment, and ideally we don't need to do it.

Reply:
It looks like a problem to be fixed in the block then :)